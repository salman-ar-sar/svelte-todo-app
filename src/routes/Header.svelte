<script>
	import { onMount } from 'svelte';
	import { typewriter } from './transitions';

	const messages = [
		'Welcome to my todo app!',
		'This site is made using just Svelte.',
		'All the transitions and animations are using Svelte helpers!',
		'There are no external libraries used for transitions.',
		'Styling is done using Tailwind CSS and Daisy UI.',
		'API routes used for the persisted Todos are also from Svelte',
	];

	let i = 0;

	onMount(() => {
		const interval = setInterval(() => {
			i += 1;
			i %= messages.length;
		}, 2500);

		return () => {
			clearInterval(interval);
		};
	});
</script>

{#key i}
	<h1 in:typewriter={{ speed: 10 }} class="min-h-12 text-3xl font-bold underline text-cyan-200">
		{messages[i] || ''}
	</h1>
{/key}
