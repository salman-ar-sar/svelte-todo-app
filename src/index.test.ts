import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/svelte';
import Header from './routes/Header.svelte';

describe('Tests Header', () => {
	it('mounts', () => {
		const { container } = render(Header);
		expect(container).toBeTruthy();
		expect(container.innerHTML).toMatchSnapshot();
	});
});
