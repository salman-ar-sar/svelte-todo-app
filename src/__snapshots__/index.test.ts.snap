// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests Header > mounts 1`] = `"<div><h1 class=\\"min-h-12 text-3xl font-bold underline text-cyan-200\\">Welcome to my todo app!</h1><!--<Header>--></div>"`;
